#Amy Sylla 

### Note
* Instructions have been included for each segment. You do not have to follow them exactly, but they are included to help you think through the steps.

In [1]:
# Dependencies and Setup
import pandas as pd

# File to Load (Remember to Change These)
file_to_load = "Resources/purchase_data.csv"

# Read Purchasing File and store into Pandas data frame
purchase_data = pd.read_csv(file_to_load)
purchase_data.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

## Player Count

* Display the total number of players


In [2]:
totalplayers = purchase_data["SN"].count()
totalplayers

780

## Purchasing Analysis (Total)

* Run basic calculations to obtain number of unique items, average price, etc.


* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame


In [3]:

totalrevenue = purchase_data["Price"].sum()
totalrevenue

totalpurchases = purchase_data["Purchase ID"].count()
totalpurchases
uniqueitems = len(purchase_data["Item ID"].unique())
uniqueitems

averageprice1 = purchase_data["Price"].mean()
averageprice1

averageprice2 = totalrevenue/totalpurchases
averageprice2

purchase_analysis = pd.DataFrame([{"Number of Unique Items": uniqueitems, "Average Price": averageprice1,
                                      "Number of Purchases": totalpurchases, "Total Revenue": totalrevenue}])
purchase_analysis["Average Price"] = purchase_analysis["Average Price"].map("${:,.2f}".format)
purchase_analysis["Total Revenue"] = purchase_analysis["Total Revenue"].map("${:,.2f}".format)
purchase_analysis

# reorganazing the column order for summary dataframe
org_purchase_analysis = purchase_analysis[["Number of Unique Items","Average Price","Number of Purchases", "Total Revenue" ]]
org_purchase_analysis

Number of Unique Items Average Price  Number of Purchases Total Revenue
0                     179         $3.05                  780     $2,379.77

## Gender Demographics

* Percentage and Count of Male Players




* Percentage and Count of Female Players


* Percentage and Count of Other / Non-Disclosed




In [4]:
gender = pd.DataFrame(purchase_data["Gender"].value_counts()) 
gender

percentageofplayers = (purchase_data["Gender"].value_counts()/totalplayers)*100
percentageofplayers

gender["Percentage of Players"] = percentageofplayers 
gender["Percentage of Players"] = gender["Percentage of Players"].map("{:,.2f}%".format)
gender

gender = gender.rename(columns={"Gender":"Total Count"})
gender

Total Count Percentage of Players
Male                           652                83.59%
Female                         113                14.49%
Other / Non-Disclosed           15                 1.92%


## Purchasing Analysis (Gender)

* Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. by gender




* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame

In [5]:
# use groupby to separate values
gendergroupedpurchased = purchase_data.groupby(["Gender"])


#total purchase
gendertotalpurchase= gendergroupedpurchased["Price"].sum()
finalgendertotalpurchasevalue = gendertotalpurchase.map("${:,.2f}".format)

# average purchase 
avgpurchaseprice = gendergroupedpurchased["Price"].mean()
totalavgpurchaseprice = avgpurchaseprice.map("${:,.2f}".format)

# normalized totals
normalizedtotals = gendertotalpurchase/gendergroupedpurchased["Purchase ID"].count()
finalnormalizedtotals = normalizedtotals.map("${:,.2f}".format)


# new dataframe
finalgenderpurchased = pd.DataFrame(gendergroupedpurchased["Purchase ID"].count())
finalgenderpurchased["Average Purchase Price"] = totalavgpurchaseprice  
finalgenderpurchased["Total Purchase Value"] = finalgendertotalpurchasevalue 
finalgenderpurchased["Normalized Totals"] = finalnormalizedtotals 
finalgenderpurchased

summarygenderpurchased = finalgenderpurchased.rename(columns={"Purchase ID":"Purchase Count"})
summarygenderpurchased

Purchase Count Average Purchase Price  \
Gender                                                         
Female                            113                  $3.20   
Male                              652                  $3.02   
Other / Non-Disclosed              15                  $3.35   

                      Total Purchase Value Normalized Totals  
Gender                                                        
Female                             $361.94             $3.20  
Male                             $1,967.64             $3.02  
Other / Non-Disclosed               $50.19             $3.35

## Age Demographics

* Establish bins for ages


* Categorize the existing players using the age bins. Hint: use pd.cut()


* Calculate the numbers and percentages by age group


* Create a summary data frame to hold the results


* Optional: round the percentage column to two decimal points


* Display Age Demographics Table


In [6]:
#cut by age
bin = [0, 9.90, 14.90, 19.90, 24.90, 29.90, 34.90, 39.90, 105]
groupnames = ["<10", "10-14", "15-19", "20-24", "25-29", "30-34", "35-39", "40+"]

grpage = purchase_data
grpage["Age Summary"] = pd.cut(grpage["Age"], bin, labels=groupnames)
grpage = grpage.groupby("Age Summary")
grpage.count()

# new table
summaryage= pd.DataFrame(grpage.count())
 

# calculations
summaryage["Purchase ID"] = (summaryage["Purchase ID"]/totalplayers)*100 

# formating 
summaryage["Purchase ID"] = summaryage["Purchase ID"].map("{:,.2f}%".format)
summaryage= summaryage[["Purchase ID","SN"]]
finalgrpage = summaryage.rename(columns={"Purchase ID":"Percentage of Players", "SN":"Total Count"})
finalgrpage

Percentage of Players  Total Count
Age Summary                                   
<10                         2.95%           23
10-14                       3.59%           28
15-19                      17.44%          136
20-24                      46.79%          365
25-29                      12.95%          101
30-34                       9.36%           73
35-39                       5.26%           41
40+                         1.67%           13

## Purchasing Analysis (Age)

* Bin the purchase_data data frame by age


* Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. in the table below


* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame

In [7]:
analysisage = pd.DataFrame(grpage["Purchase ID"].count())

totalvalueage = grpage["Price"].sum()
ftotalvalueage = totalvalueage.map("${:,.2f}".format)


avgage = grpage["Price"].mean()
tavgage = avgage.map("${:,.2f}".format)

normalizedtotalsage = totalpurchasevalueage/grpage["Purchase ID"].count()
tnormalizedtotalsage = normalizedtotalsage.map("${:,.2f}".format)


analysisage["Average Purchase Price"] = tavgage  
analysiage["Total Purchase Value"] = ftotalvalueage 
analysisage["Normalized Totals"] = tnormalizedtotalsage 

analysisage = analysiage.rename(columns={"Purchase ID":"Purchase Count"})
analysisage

Purchase Count Average Purchase Price Total Purchase Value  \
Age Ranges                                                              
<10                    23                  $3.35               $77.13   
10-14                  28                  $2.96               $82.78   
15-19                 136                  $3.04              $412.89   
20-24                 365                  $3.05            $1,114.06   
25-29                 101                  $2.90              $293.00   
30-34                  73                  $2.93              $214.00   
35-39                  41                  $3.60              $147.67   
40+                    13                  $2.94               $38.24   

           Avg Total Purchase per Person  
Age Ranges                                
<10                                $4.54  
10-14                              $3.76  
15-19                              $3.86  
20-24                              $4.32  
25-29                              $3.81  
30-34                              $4.12  
35-39                              $4.76  
40+                                $3.19

## Top Spenders

* Run basic calculations to obtain the results in the table below


* Create a summary data frame to hold the results


* Sort the total purchase value column in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the summary data frame



In [7]:
grpSN = purchase_data.groupby("SN")
grpSN.count()

analysisSPENDOR = pd.DataFrame(grpSN["Purchase ID"].count())

# Get Total purchase value by SN
totalSN = grpSN["Price"].sum()


# Get Average purchase price by SN
avgSN = grpSN["Price"].mean()
tavgSN = avgSN.map("${:,.2f}".format)

# Organize summary Top Spender data, get all columns to organized Data Frame, add needed columns to it
analysisSPENDOR["Average Purchase Price"] = tavgSN
analysisSPENDOR["Total Purchase Value"] = totalSN 


SNpurchased = analysisSPENDOR.rename(columns={"Purchase ID":"Purchase Count"})
TOP5spendors= SNpurchased.sort_values("Total Purchase Value", ascending=False)

ftotal = totalSN.map("${:,.2f}".format)
TOP5spendors["Total Purchase Value"] = ftotal
TOP5spendors.head()

Purchase Count Average Purchase Price Total Purchase Value
SN                                                                     
Lisosia93                 5                  $3.79               $18.96
Idastidru52               4                  $3.86               $15.45
Chamjask73                3                  $4.61               $13.83
Iral74                    4                  $3.40               $13.62
Iskadarya95               3                  $4.37               $13.10

## Most Popular Items

* Retrieve the Item ID, Item Name, and Item Price columns


* Group by Item ID and Item Name. Perform calculations to obtain purchase count, item price, and total purchase value


* Create a summary data frame to hold the results


* Sort the purchase count column in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the summary data frame



In [9]:
grpitem = purchase_data.groupby(["Item ID", "Item Name"])

analysisITEM = pd.DataFrame(grpitem["Purchase ID"].count())
totalITEM = grpitem["Price"].sum()
ftotalITEM = totalITEM.map("${:,.2f}".format)
purchaseprice = grpitem["Price"].mean()
fpurchaseprice = purchaseprice.map("${:,.2f}".format)
analysisITEM["Item Price"] = fpurchaseprice
analysisITEM["Total Purchase Value"] = ftotalITEM
SUMITEM = analysisITEM.rename(columns={"Purchase ID":"Purchase Count"})
TOP5ITEMS = SUMITEM.sort_values("Purchase Count", ascending=False)
TOP5ITEMS.head()

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
132     Persuasion                                                 9   
108     Extraction, Quickblade Of Trembling Hands                  9   

                                                     Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
132     Persuasion                                        $3.22   
108     Extraction, Quickblade Of Trembling Hands         $3.53   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
132     Persuasion                                                 $28.99  
108     Extraction, Quickblade Of Trembling Hands                  $31.77

## Most Profitable Items

* Sort the above table by total purchase value in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the data frame



In [13]:
SUMITEM["Total Purchase Value"] = grpitem["Price"].sum()
TOP5items = SUMITEM.sort_values("Total Purchase Value", ascending=False)
finaltotalitem = totalITEM.map("${:,.2f}".format)
TOP5items["Total Purchase Value"] = ftotalITEM
TOP5items.head()

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
82      Nirvana                                                    9   
145     Fiery Glass Crusader                                       9   
103     Singed Scalpel                                             8   

                                                     Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
82      Nirvana                                           $4.90   
145     Fiery Glass Crusader                              $4.58   
103     Singed Scalpel                                    $4.35   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
82      Nirvana                                                    $44.10  
145     Fiery Glass Crusader                                       $41.22  
103     Singed Scalpel                                             $34.80

We can make the conclusion that :
 
 - Out of the 780 player base, there is more male players (83.59%) than female players (14%~), female players spend a little bit more money on average purchases ($3.20 compared to $3.02) 
 - The core of the player base is between 20-24 year old. The smallest group is 40+ years old.
 - The most profitable item is  "Final Critic" (13 purchase counts for $4.61) however "Oathbreaker, Last Hope of the Breaking Storm" is really close behind it (12 purchase counts for $4.23).
 